File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.12 Funkcją o prototypie bool PAR(int A, int B, int OP=0).py
def silnia(N):
    result = 1
    for i in range(2, N + 1):
        result *= i
    return result

if __name__ == "__main__":
    x = int(input("Podaj liczbę całkowitą x: "))
    n = int(input("Podaj liczbę całkowitą n: "))
    suma = sum((x ** i) / silnia(i) for i in range(n + 1))
    print("Wynik:", suma)

File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.12 Funkcją o prototypie bool PAR(int A, int B, int OP=0).cpp
#include <iostream>
#include <cmath>

unsigned long long silnia(int N) {
    unsigned long long result = 1;
    for (int i = 2; i <= N; ++i) {
        result *= i;
    }
    return result;
}

int main() {
    int x, n;
    std::cout << "Podaj liczbê ca³kowit¹ x: ";
    std::cin >> x;
    std::cout << "Podaj liczbê ca³kowit¹ n: ";
    std::cin >> n;
    double suma = 0;
    for (int i = 0; i <= n; ++i) {
        suma += pow(x, i) / silnia(i);
    }
    std::cout << "Wynik: " << suma << std::endl;
    return 0;
}
